pkg/kube/kclient/client.go

```diff
@@ -90,6 +90,43 @@ func (n *informerClient[T]) Start(stopCh <-chan struct{}) {
 	n.startInformer(stopCh)
 }
 
+// internalIndex is the type we use to store a Kubernetes set of index, as well as the key to the index we care about.
+// Users should not interact with this directly, and should instead use CreateIndex.
+type internalIndex struct {
+	key     string
+	indexer cache.Indexer
+}
+
+func (i internalIndex) Lookup(key string) []interface{} {
+	res, err := i.indexer.ByIndex(i.key, key)
+	if err != nil {
+		// This should only happen if the key does not exist which should be impossible
+		log.Fatalf("index lookup failed: %v", err)
```

Comment:
why i.indexer.ByIndex cannot return not exist error, where do you protect it

Reply:
The only time it returns an error is if the index doesn't exist. We know not must exist because we were the ones that created it